cmake/options.cmake

```diff
@@ -90,6 +90,12 @@ if (WITH_DCACHE_BUFFERS)
   add_definitions(-DVIRTIO_CACHED_BUFFERS)
 endif (WITH_DCACHE_BUFFERS)
 
+option (WITH_DCACHE_RSC_TABLE "Build with resource table cache operations enabled" OFF)
+
+if (WITH_DCACHE_RSC_TABLE)
+  add_definitions(-DVIRTIO_CACHED_RSC_TABLE)
```

Comment:
Well.. this is tricky - why we have 2 options now? Why don't we use only one?
I believe it's better to treat them separately because each has it's own memory assigned in the dts.

Reply:
We can also assume that most of the project would enable the cache for vrings, buffers and resource table.
To simplify the use, perhaps it would be nice to introduce an option to enable cache for all memories.

```
option (WITH_DCACHE "Build with all cache operations enabled" OFF)

option (WITH_DCACHE_VRINGS "Build with vrings cache operations enabled" OFF)
if (WITH_DCACHE_VRINGS OR WITH_DCACHE)
  add_definitions(-DVIRTIO_CACHED_VRINGS)
endif (WITH_DCACHE_VRINGS)
option (WITH_DCACHE_BUFFERS "Build with vrings cache operations enabled" OFF)
if (WITH_DCACHE_BUFFERS OR WITH_DCACHE)
  add_definitions(-DVIRTIO_CACHED_BUFFERS)
endif (WITH_DCACHE_BUFFERS)

option (WITH_DCACHE_RSC_TABLE "Build with resource table cache operations enabled" OFF)

if (WITH_DCACHE_RSC_TABLE OR WITH_DCACHE)
  add_definitions(-DVIRTIO_CACHED_RSC_TABLE)
endif (WITH_DCACHE_RSC_TABLE)

```

i just also notice an error in description (It a bonus that can be fixed later, as @carlocaione  said)
```
- option (WITH_DCACHE_BUFFERS "Build with vrings cache operations enabled" OFF)
+ option (WITH_DCACHE_BUFFERS "Build with buffers cache operations enabled" OFF)
```

Reply:
@arnopo I agree. We can assume for whole platform either cache is enabled or not. Anyway, There are many configurations already available for libmetal. It's good idea to combine all three in one. However, we can do that cleanup in separate PR, doesn't have to be as part of this one.

Reply:
@arnopo @carlocaione @TanmayShah-xilinx in this PR I'll just add a new option - VIRTIO_CACHED_RSC_TABLE.

I'll follow up with the other suggestions in a separate PR - updates & clean-up with:
- add a new option that enables all 3 cache options - RINGS, BUFFERS, RSC_TABLE;
- s/vrings/buffers for WITH_DCACHE_BUFFERS option;
- use defines for cache functions for BUFFERS, the same as for RINGS and RSC_TABLE.